src/Components/Components/src/NavigationManager.cs

```diff
@@ -177,6 +196,19 @@ protected virtual void NavigateToCore([StringSyntax(StringSyntaxAttribute.Uri)]
     public virtual void Refresh(bool forceReload = false)
         => NavigateTo(Uri, forceLoad: true, replace: true);
 
+    /// <summary>
+    /// Handles setting the NotFound state.
+    /// </summary>
+    public virtual void NotFound() => NotFoundCore();
+
+    /// <summary>
+    /// Handles setting the NotFound state.
+    /// </summary>
+    protected virtual void NotFoundCore()
+    {
+        _notFound?.Invoke(this, new EventArgs());
+    }
```

Comment:
Do we need this method to be public? None of our current platforms overrides it, does it?